File: Introduction to Data Structures: Binary Tree/1. Traverse A Tree/2. Binary Tree In-Order Traversal/1. Recursive Method/BinaryTreeInorderTraversalRecursive.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:

        output = []

        def traverseInorder(node):

            if node == None:

                return

            traverseInorder(node.left)

            output.append(node.val)

            traverseInorder(node.right)

        traverseInorder(root)

        return output

File: Introduction to Data Structures: Binary Tree/1. Traverse A Tree/2. Binary Tree In-Order Traversal/1. Recursive Method/BinaryTreeInorderTraversalRecursive.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

 class Solution {
    
    // In order traversal : left -> root -> right
    List<Integer> output = new ArrayList<>();
    
    public List<Integer> inorderTraversal(TreeNode root) {
        
        // base case: if left child node is null return output
        if (root == null) {
            return output;
        }
        
        // get value of the left child
        inorderTraversal(root.left);
        
        // get value of the root node
        output.add(root.val);
        
        // get value of the right node
        inorderTraversal(root.right);
        
        return output;
    }
}